### ISSUE TITLE
The Drone Racing League Simulator (641780)

### ISSUE THREAD

**Issue by tenten8401** [ValveSoftware/Proton#2910]
# Compatibility Report
- Name of the game with compatibility issues: DRL Simulator
- Steam AppID of the game: 641780

## System Information
- GPU: RX 580
- Driver/LLVM version: Mesa 19.0.8
- Kernel version: 5.1.18
- Link to full system information report as [Gist](https://gist.github.com/): https://gist.github.com/tenten8401/71255b32a5d7af25f33a645bf36de5f0
- Proton version: 4.2-9

## I confirm:
- [X] that I haven't found an existing compatibility report for this game.
- [X] that I have checked whether there are updates for my system available.

## Symptoms
Just starts to a black screen, nothing I've tried can get it past.
[steam-641780.log](https://github.com/ValveSoftware/Proton/files/3434055/steam-641780.log)

## Reproduction
Start the game

**Comment by kisak-valve** [comment]
Looks like this game would benefit from support for Media Foundation maturing.

**Comment by baracoder** [comment]
(proton 5.0-3) I can confirm that it starts up fine with `<Link removed by moderator>` installed. There is another problem: It detects, a gampad model name, but does not recognize inputs.

**Comment by kisak-valve** [comment]
Hello @baracoder, the workaround you linked is legally problematic and has been removed.

**Comment by tenten8401** [comment]
Yeah I noticed the gamepad thing too. Interestingly enough it will detect a PS4 controller but none of the OpenTX-based joysticks, even though they show up in the wine joystick configuration.

**Comment by sfjuocekr** [comment]
Can confirm no inputs registered, game works fine otherwise.

I've even gotten it to recognize the transmitter as an xbox gamepad through SteamInput, but I'm kinda stumped why it pretends to be in keyboard/mouse mode. I can move the cursor and even items in game, but I can't map anything.

According to the developer of Rewired this might have to do with raw input:

"I had to work with Valve to get Raw Input working in Proton, their Wine fork. They had to implement a bunch of Raw Input and HID functionality that was not built into Wine. I have no idea if they ported that back to the main branch."

If I go back through changes, these changes "should" have been included since 4.7.

For the record, I checked joystick support with wine/proton using PicaSim and DCL which seems to work fine.

**Comment by outerSpaceToast** [comment]
I also can confirm that the input mapping does not work. The input values change when the sticks are moved, but it feels like random numbers (uint8<->uint16?). There is definitely a problem in the input aquisition.

@sfjuocekr : Do you know which commits the Rewired developer meant? Maybe I can have a look at them.

**Comment by sfjuocekr** [comment]
I think this is the one: https://github.com/Frogging-Family/wine-tkg-git/blob/master/wine-tkg-git/wine-tkg-patches/proton/proton-rawinput.patch

I'm not even getting any input values, they are all zeroed.

**Comment by outerSpaceToast** [comment]
The mentioned patch seems to be implemented partly. In my opinion some statements are not implemented (However, I can't assess, whether they are important).
It seems that the latest version of Proton-GE (6.1) has this patch included.
But I was not successful to run it with Proton-GE 6.1 as I only get a black screen again. Installing `<Workaround removed by moderator>` files didn't help.

**Comment by kisak-valve** [comment]
Hello @outerSpaceToast, the workaround you mentioned is legally problematic and has been removed.

**Comment by kisak-valve** [comment]
Drone Racing League game not recongnizing the throttle input

Issue transferred from https://github.com/ValveSoftware/Proton/issues/5249.
@mrlinux11 posted on 2021-10-25T13:04:11:

I am using a Radiomaster TX16s remote as a joystick, under Windows it works fine, but under Linux using Proton the throttle is not recognized but all other inputs are.  I tested using https://gamerpad-test.com and it recognizing all the inputs.

**Comment by AquaMorph** [comment]
This exact issue is also a problem on another FPV simulator called AI Drone Simulator. It behaves exactly the same for me as Drone Racing League. Liftoff has native Linux support and the controls work fine. I used the DJI FPV Remote for my testing.

**Comment by grahamjenson** [comment]
This is still an issue. Running Liftoff with the tx16s controller under native linux works, but via proton the inputs are mismapped so the throttle doesnt work.

**Comment by sfjuocekr** [comment]
This has nothing todo with the controller used, but everything with the way DRL seems to handle input.
